File: articles/causal-algorithms-and-intervention-modeling/haskell/InterventionDecision.hs
-- Compact Haskell example: typed intervention comparison.
data Decision = Act | Monitor deriving (Show, Eq)

decide :: Double -> Double -> Decision
decide threshold score = if score >= threshold then Act else Monitor

effect :: Double -> Double -> Double
effect baseline intervention = intervention - baseline

main :: IO ()
main = do
  let baseline = 0.42
      intervention = 0.57
      score = 0.53
  print (effect baseline intervention)
  print (decide 0.55 score, decide 0.50 score)

File: articles/causal-algorithms-and-intervention-modeling/java/InterventionDecision.java
public class InterventionDecision {
    static String decision(double score, double threshold) {
        return score >= threshold ? "act" : "monitor";
    }

    public static void main(String[] args) {
        double baseline = 0.42;
        double intervention = 0.57;
        System.out.printf("estimated_effect=%.6f%n", intervention - baseline);
        System.out.println("baseline_decision=" + decision(0.53, 0.55));
        System.out.println("new_threshold_decision=" + decision(0.53, 0.50));
    }
}
